js/services.js

```diff
@@ -0,0 +1,166 @@
+/* global ServiceConfig, Promise, Messenger */
+
+'use strict';
+
+(function(exports) {
+  /**
+   * This module is responsible to create and run services.
+   */
+  var Services = function() {};
+  Services.prototype = {
+
+    _dispatchReady: function() {
+      // With all important event handlers in place, we can now notify
+      // Gecko that we're ready for certain system services to send us
+      // messages (e.g. the radio).
+      // Note that shell.js starts listen for the mozContentEvent event at
+      // mozbrowserloadstart, which sometimes does not happen till
+      // window.onload.
+      var evt = new CustomEvent('mozContentEvent', {
+        bubbles: true, cancelable: false,
+        detail: { type: 'system-message-listener-ready'}
+      });
+      window.dispatchEvent(evt);
+    },
+
+    handleBrowserLoaded: function(instance) {
```

Comment:
It's better go with basic doc that what is window/app for -> or daemon/module/app for , ex:

```
there are 3 types of container provided by system: daemon, module, app
* daemon: module runs without visual presentation
* module: embedded iframe, can communicate with system with postMessage
* app: module runs in different process
```